src/ray/gcs/gcs_server/test/gcs_health_check_manager_test.cc

```diff
@@ -71,6 +71,7 @@ class GcsHealthCheckManagerTest : public ::testing::Test {
     auto channel = grpc::CreateChannel("localhost:" + std::to_string(port),
                                        grpc::InsecureChannelCredentials());
     server->Run();
+    server->GetServer().GetHealthCheckService()->SetServingStatus(node_id.Hex(), true);
     servers.emplace(node_id, server);
```

Comment:
per our sync: let's add a few unit tests to verify the behavior when service_id is not set, different, same, for both client and server.